### DISCUSSION TITLE
Add support for AWS IAM role auth type

### DISCUSSION THREAD

**Discussion by gtsoukasaws** [grafana/grafana-infinity-datasource#902] [Ideas]
For Authentication, add an option to use AWS IAM role based authentication, so that no secrets must be managed in Grafana by the user. This would function similar to the Amazon [CloudWatch data source](https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/).

**Comment by johan1252** [comment]
This would be very useful, and something other plugins already support.

**Comment by MarcBridner** [comment]
For instance credentials using the default credential chain, everything that needs to be changed on the Go backend is to add 
```
AWSAuthTypeDefault AWSAuthType = "default"
```
to `pkg/models/settings.go`. 

A corresponding UI change also has to be made, to allow setting it in options. We were experimenting with it locally because we needed to connect to the raw endpoint for AWS OpenSearch Serverless, to run aggregations.

Unfortunately this particular endpoint also requires [SHA256 encoding the body](https://github.com/opensearch-project/opensearch-go/blob/main/signer/aws/aws.go#L126-L132) and we couldn't figure out a good way to make this generic enough to submit a PR
